src/__tests__/template-literal.test.ts

```diff
@@ -0,0 +1,595 @@
+// @ts-ignore TS6133
+import { expect, test } from "@jest/globals";
```

Comment:
One other thing I ran into: number representations of exponents, e.g. `2e5` will evidently not work with `addInterpolatedPosition(z.number())`. I think this is probably okay, and its inherent with the limitation of treating everything as a string first and then applying the regex. However it _is_ a valid TS number, would it be worth it to include as a test to explicitly state that we're aware of it or TODO/FIXME?

Reply:
Great catch!

By definition, everything that has typeof number at runtime, expect for NaN (since zod has special treatment for it), should be supported. Let's add support for this.